### PR TITLE
Favor Observable.defer and AsyncSubject over Observable.create

### PR DESCRIPTION
**Pull request by VictorAlbertos**
Address #137

### CODE REVIEW

File: fuel-rxjava/src/main/kotlin/com/github/kittinunf/fuel/rx/RxFuel.kt [hunk 1]
```diff
@@ -19,21 +18,21 @@ fun Request.rx_responseString(charset: Charset = Charsets.UTF_8) = rx_response(S
...
 
                     is Result.Failure -> {
-                        subscriber.onError(result.error)
+                        source.onError(result.error)
                     }
                 }
             }
-
-            subscriber.add(createRequestSubscription(this))
+            source.doOnSubscribe { this.cancel() }
```
Comment (kittinunf) [inline fuel-rxjava/src/main/kotlin/com/github/kittinunf/fuel/rx/RxFuel.kt]: Shouldn't this be `doOnUnSubscribe`?

File: fuel-rxjava/src/main/kotlin/com/github/kittinunf/fuel/rx/RxFuel.kt [hunk 2]
```diff
@@ -43,31 +42,20 @@ fun Request.rx_string(charset: Charset = Charsets.UTF_8) = rx_result(StringDeser
...
```
Comment (kittinunf) [inline fuel-rxjava/src/main/kotlin/com/github/kittinunf/fuel/rx/RxFuel.kt]: Same here.

**Comment by VictorAlbertos** [comment]
You're right. Sorry about that. By the way, the build is failing:

> The android gradle plugin version 2.3.0-beta3 is too old, please update to the latest version.

Do you want me to bump the gradle version?

**Comment by kittinunf** [comment]
I have opened #139 to fix this! Could you please rebase this PR again master as soon as #139 get merged?

**Comment by VictorAlbertos** [comment]
Sure!

**Comment by VictorAlbertos** [comment]
Done!

**Comment by kittinunf** [comment]
Thanks!!!!!

**Pull request merged**